### PR TITLE
Increase minimum compatible version to 2.6.0

### PR DESCRIPTION
**Pull request by joejuzl**
**Proposed changes**:
Increase minimum compatible version to 2.6.0

**Status (please check what you already did)**:
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)

### CODE REVIEW

File: changelog/8498.removal.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Update MINIMUM_COMPATIBLE_VERSION to "2.6.0".
```
Comment (wochinge) [inline changelog/8498.removal.md:1]: Can we please phrase that more user-facing? Users should know that this means that they have to retrain the model with 2.6 in order to run with 2.6

**Review by wochinge — APPROVED** [review]
💯

**Comment by wochinge** [comment]
Re the failing tests -> I think you have to do the change right after running `make release` and not as a separate PR

**Comment by joejuzl** [comment]
> Re the failing tests -> I think you have to do the change right after running make release and not as a separate PR

haha yes ok - that makes sense. I will merge this branch into the release branch.

**Comment by joejuzl** [comment]
Closing as now part of #8620

**Pull request closed without merging**